src/compiler/crystal/codegen/link.cr

```diff
@@ -100,7 +100,7 @@ module Crystal
 
   class Program
     def object_extension
-      has_flag?("windows") ? ".obj" : ".o"
+      has_flag?("windows") ? ".obj" : has_flag?("wasm32") ? ".wasm" : ".o"
```

Comment:
what do you think about restructuring this as `if/elsif/else` statements instead? The nested ternaries are a bit too terse and difficult to read IMO